### PR TITLE
Revamped IPC and IDL

### PR DESCRIPTION
**Pull request by sleepy-monax**

### CODE REVIEW

File: sources/libs/ipc/hooks.h [hunk 1]
```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include <ipc/component.h>
+
+typedef enum
+{
+    IDL_HANDLER_REQ_RESP,
+    IDL_HANDLER_NIL_RESP,
+    IDL_HANDLER_REQ_NIL,
+    IDL_HANDLER_NIL_NIL,
```
Comment (Supercip971) [inline sources/libs/ipc/hooks.h:10]: why not put NIL_NIL to the top so it is 0 and not 3 ?

File: sources/libs/stdc/math/lrint.c [hunk 2]
```diff
@@ -13,7 +13,7 @@ C99 is a bit vague about whether inexact exception is
 allowed to be raised when invalid is raised.
 (F.9 explicitly allows spurious inexact exceptions, F.9.6.5
 does not make it clear if that rule applies to lrint, but
-IEEE 754r 7.8 seems to forbid spurious inexact exception in
+IEEE 754r 7.8 seems to foridl spurious inexact exception in
```
Comment (Supercip971) [inline sources/libs/stdc/math/lrint.c:16]: rip

**Review by Supercip971 — APPROVED** [review]

File: sources/libs/ipc/hooks.h (hunk 1 above)
Comment (sleepy-monax) [reply sources/libs/ipc/hooks.h]: Fix

File: sources/libs/stdc/math/lrint.c (hunk 2 above)
Comment (sleepy-monax) [reply sources/libs/stdc/math/lrint.c:16]: lol

**Review by Supercip971 — APPROVED** [review]

**Pull request merged**